third_party/blink/renderer/modules/cobalt/h5vcc_experiments/h_5_vcc_experiments.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "third_party/blink/renderer/modules/cobalt/h5vcc_experiments/h_5_vcc_experiments.h"
+
+#include <string>
+
+#include "base/values.h"
+#include "cobalt/browser/cobalt_experiment_names.h"
+#include "cobalt/browser/h5vcc_experiments/public/mojom/h5vcc_experiments.mojom-blink.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise.h"
+#include "third_party/blink/renderer/bindings/core/v8/script_promise_resolver.h"
+#include "third_party/blink/renderer/bindings/core/v8/v8_union_boolean_long_string.h"
+#include "third_party/blink/renderer/bindings/modules/v8/v8_experiment_configuration.h"
+#include "third_party/blink/renderer/core/frame/local_dom_window.h"
+
+namespace blink {
+
+H5vccExperiments::H5vccExperiments(LocalDOMWindow& window)
+    : ExecutionContextLifecycleObserver(window.GetExecutionContext()),
+      remote_h5vcc_experiments_(window.GetExecutionContext()) {}
+
+void H5vccExperiments::ContextDestroyed() {}
+
+ScriptPromise H5vccExperiments::setExperimentState(
+    ScriptState* script_state,
+    const ExperimentConfiguration* experiment_configuration,
+    ExceptionState& exception_state) {
```

Comment:
@yell0wd0g what is ExceptionState and what's it used for?

Reply:
It allows a method to throw an Exception, e.g. if the Mojo pipe is
not connected. It's a bit redundant with a Promise rejection, since
both can be used to signal a problem, but an exception is asynchronous.
Note that it's a Chromium-ism https://chromium.googlesource.com/chromium/src/+/HEAD/third_party/blink/renderer/bindings/IDLExtendedAttributes.md#raisesexception and it appears due to the IDL
annotation `RaisesException` added in this PR/CL.